src/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -293,16 +291,39 @@ public SAMFileWriter makeSAMOrBAMWriter(final SAMFileHeader header, final boolea
         return makeBAMWriter(header, presorted, outputFile);
     }
 
+    /**
+     *
+     * Create a SAM, BAM or CRAM writer based on examination of the outputFile extension.
+     * For CRAM files, read names and tags are preserved.
```

Comment:
What does it mean that for CRAM files read names and tags are preserved?  Are they changed in SAM/BAM files?